docker/Transformation-Engine/src/test/kotlin/net/nhsd/fhir/converter/transformer/allergyIntoleranceEnd.kt

```diff
@@ -0,0 +1,122 @@
+package net.nhsd.fhir.converter.transformer
+
+import ca.uhn.fhir.context.FhirVersionEnum
+import net.javacrumbs.jsonunit.assertj.JsonAssert
+import org.assertj.core.api.Assertions.assertThat
+import org.hl7.fhir.r4.model.DateTimeType
+import org.hl7.fhir.r4.model.StringType
+import org.hl7.fhir.r4.model.MedicationRequest
+import org.hl7.fhir.dstu3.model.AllergyIntolerance as R3AllergyIntolerance
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.Arguments
+import org.junit.jupiter.params.provider.MethodSource
+import org.junit.jupiter.params.provider.ValueSource
+import org.springframework.http.MediaType
+import java.util.stream.Stream
+import org.hl7.fhir.dstu3.model.Extension as R3Extension
+import org.hl7.fhir.r4.model.AllergyIntolerance as R4AllergyIntolerance
+
+
+internal class AllergyIntoleranceEndTest {
+    @ParameterizedTest
+    @ValueSource(strings = [CARECONNECT_ALLERGY_INTOLERANCE_END_URL, UKCORE_ALLERGY_INTOLERANCE_END_URL])
+    internal fun `it should create extension with the associated ukcore url`(extUrl: String) {
+        // Given
+        val r3Extension = R3Extension().apply {
+            url = extUrl
+        }
+
+        val r4Resource = R4AllergyIntolerance()
+
+        // When
+        allergyIntoleranceEnd(r3Extension, r4Resource)
+
+        // Then
+        val transformedExt = r4Resource.getExtensionByUrl(UKCORE_ALLERGY_INTOLERANCE_END_URL)
+        assertThat(transformedExt).isNotNull
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = [CARECONNECT_ALLERGY_INTOLERANCE_END_URL, UKCORE_ALLERGY_INTOLERANCE_END_URL])
```

Comment:
i think you would only need careconnect url as value source here? And same for test below